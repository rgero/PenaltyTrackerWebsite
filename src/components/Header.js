import React from 'react';
import { Link } from 'react-router-dom';


const Header = () => (
    <header className="header">
        <div className="header__content">
            <Link className="header__title" to="/" exact="true"><h1>The Penalty Tracker</h1></Link>
        </div>
    </header>
);

export default Header;