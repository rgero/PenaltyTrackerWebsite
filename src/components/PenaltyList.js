import React from 'react';
import { connect } from 'react-redux';

import _ from "lodash";

// Import React Table
import ReactTable from "react-table";
import "react-table/react-table.css";

export const PenaltyList = (props) => {
    var data = props.penalties;
    return (
        <div className="content-container">
            <ReactTable
                data={data}
                columns={[
                    {
                    Header: "Penalty",
                    columns: [
                        {
                        Header: "Name",
                        accessor: "playerName"
                        },
                        {
                            Header: "Team Name",
                            accessor: "teamName"
                        },
                        {
                            Header: "Opponent",
                            accessor: "opponentTeam"
                        },
                        {
                            Header: "Game Date",
                            accessor: "gameDate"
                        },
                        {
                            Header: "Penalty",
                            accessor: "penalty"
                        },
                        {
                            Header: "Location",
                            accessor: "homeAway",
                            Cell: props => <div> { props.value === 1 ? "Home" : "Away" } </div>
                        }
                    ]
                    }
                ]}
                defaultPageSize={10}
                className="-striped -highlight"
                />
        </div>
    )
};

const mapStateToProps = (state, props)=>{
    return {
        penalties: state
    };
}

export default connect(mapStateToProps)(PenaltyList);