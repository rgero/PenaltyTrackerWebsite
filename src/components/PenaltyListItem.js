import React from 'react';

export const PenaltyListItem = (props) => (
    <div className="list-item">
        <div className="show-for-desktop">{props.playerName}</div>
        <div className="show-for-desktop">{props.teamName}</div>
        <div className="show-for-desktop">{props.opponentTeam}</div>
        <div className="show-for-desktop">{props.penalty}</div>
        <div className="show-for-desktop">{props.gameDate}</div>
        <div className="show-for-desktop">{props.homeAway === 1 ? ( "Home" ) : ("Away") }</div>
        <div className="show-for-desktop">{props.refs}</div>
    </div> 
)

export default PenaltyListItem;