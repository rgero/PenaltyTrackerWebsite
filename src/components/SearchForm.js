import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export class SearchForm extends React.Component {

    constructor(props){
        super(props);

        //Functions
        this.onSubmit = this.onSubmit.bind(this);
        this.onTextChange = this.onTextChange.bind(this);
        this.onTeamChange = this.onTeamChange.bind(this);
        this.onSeasonChange = this.onSeasonChange.bind(this);
        this.onLocationChange = this.onLocationChange.bind(this);
        this.onDateChange = this.onDateChange.bind(this);
        this.onCalendarFocusChanged = this.onCalendarFocusChanged.bind(this);
        this.resetState = this.resetState.bind(this);

        this.state = {
            players: "",
            teams: [],
            opponents: [],
            location: "either",
            penalty: "",
            startDate: undefined,
            endDate: undefined,
            season: "Regular_18_19",
            startCalendarFocused: false,
            endCalendarFocused: false
        };
    };
    
    onSubmit(e){
        e.preventDefault();
        this.props.onSubmit({
            players: this.state.players,
            teams: this.state.teams,
            opponents: this.state.opponents,
            location: this.state.location,
            penalty: this.state.penalties,
            startDate: this.state.startDate,
            endDate: this.state.endDate,
            season: this.state.season
        })
    }

    resetState(){
        this.setState({
            players: "",
            teams: [],
            opponents: [],
            location: "either",
            penalty: "",
            startDate: undefined,
            endDate: undefined,
            season: "Regular_18_19",
            startCalendarFocused: false,
            endCalendarFocused: false
        })
    }

    onTeamChange = (key) => (e) => {
        var options = e.target.options;
        var values = [];
        for (var i = 0, l = options.length; i < l; i++) {
            if (options[i].selected) {
            values.push(options[i].value);
            }
        }
        if (key === "player"){
            this.setState({
                teams: values
            })
        } else if (key === "opponent"){
            this.setState({
                opponents: values
            })
        }
    }

    

    onSeasonChange = (e) => {
        var season = e.target.value;
        this.setState({
            season
        })
    }

    onTextChange = (key) => (e) => {
        var newValue = e.target.value;
        switch(key){
            case "name":
                this.setState({
                    players: newValue
                });
                break;
            case "penalty":
                this.setState({
                    penalty: newValue
                })
                break;
            default:
                break;
        }
    }

    onDateChange = (key) => (e) => {
        switch(key){
            case "start":
                this.setState({
                    startDate: e
                })
                break;
            case "end":
                this.setState({
                    endDate: e
                })
                break;
            default:
                break;
        }
    }

    onCalendarFocusChanged = (key) => (calendarFocused) => {
        calendarFocused = calendarFocused["focused"]
        switch(key){
            case "start":
                this.setState(() => ({startCalendarFocused: calendarFocused}));
                break;
            case "end":
                this.setState(() => ({endCalendarFocused: calendarFocused}));
                break;    
            default:
                break;
        }
    }


    onLocationChange = (e) => {
        var location = e.target.value;
        this.setState(
            {location}
        )

    }

    render() {

        let seasonList = [
            { id: "Regular_16_17", name: "2016-17 Regular"},
            { id: "Playoffs_16_17", name: "2016-17 Playoffs"},
            { id: "Regular_17_18", name: "2017-18 Regular"},
            { id: "Playoffs_17_18", name: "2017-18 Playoffs"},
            { id: "Regular_18_19", name: "2018-19 Regular"}
        ]

        let seasonOptions = seasonList.map((season)=> {
            return <option key={season.id} value={season.id}>{season.name}</option>
        })

        let teamList =  [   { id: "ANA", name: "Anaheim Ducks"},
                            { id: "ARI", name: "Arizona Coyotes"},
                            { id: "BOS", name: "Boston Bruins"},
                            { id: "BUF", name: "Buffalo Sabres"},
                            { id: "CGY", name: "Calgary Flames"},
                            { id: "CAR", name: "Carolina Hurricanes"},
                            { id: "CHI", name: "Chicago Blackhawks"},
                            { id: "COL", name: "Colorado Avalanche"},
                            { id: "CBJ", name: "Columbus Blue Jackets"},
                            { id: "DAL", name: "Dallas Stars"},
                            { id: "DET", name: "Detroit Red Wings"},
                            { id: "EDM", name: "Edmonton Oilers"},
                            { id: "FLA", name: "Florida Panthers"},
                            { id: "LAK", name: "Los Angeles Kings"},
                            { id: "MIN", name: "Minnesota Wild"},
                            { id: "MTL", name: "Montreal Canadiens"},
                            { id: "NSH", name: "Nashville Predators"},
                            { id: "NJD", name: "New Jersey Devils"},
                            { id: "NYI", name: "New York Islanders"},
                            { id: "NYR", name: "New York Rangers"},
                            { id: "OTT", name: "Ottawa Senators"},
                            { id: "PHI", name: "Philadelphia Flyers"},
                            { id: "PIT", name: "Pittsburgh Penguins"},
                            { id: "SJS", name: "San Jose Sharks"},
                            { id: "STL", name: "St Louis Blues"},
                            { id: "TBL", name: "Tampa Bay Lightning"},
                            { id: "TOR", name: "Toronto Maple Leafs"},
                            { id: "VAN", name: "Vancouver Canucks"},
                            { id: "VGK", name: "Vegas Golden Knights"},
                            { id: "WSH", name: "Washington Capitals"},
                            { id: "WPG", name: "Winnipeg Jets"}
                        ]
        
        let teamOptions = teamList.map((team)=> {
            return <option key={team.id} value={team.id}>{team.name}</option>
        })

        return (
            <div className="content-container">
                <form className="form" onSubmit={this.onSubmit}>
                    <div className="form__input">
                        <label>Season</label>
                        <select className="select" onChange={this.onSeasonChange} value={this.state.season}>
                            {seasonOptions}
                        </select>
                    </div>
                    <div className="form__input">
                        <label>Player Name</label>
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Name"
                            autoFocus
                            value={this.state.players}
                            onChange={this.onTextChange("name")}
                        />
                    </div>
                    <div className="form__input">
                        <label>Penalty</label>
                        <input
                            type="text"
                            className="text-input"
                            placeholder="Penalty"
                            autoFocus
                            value={this.state.penalty}
                            onChange={this.onTextChange("penalty")}
                        />
                    </div>
                    <div className="form__input">
                        <label>Player's Team</label>
                        <select multiple className="selectTeams" onChange={this.onTeamChange("player")} value={this.state.teams}>
                            {teamOptions}
                        </select>
                    </div>
                    <div className="form__input">
                        <label>Opponent Team</label>
                        <select multiple className="selectTeams" onChange={this.onTeamChange("opponent")} value={this.state.opponents}>
                            {teamOptions}
                        </select>
                    </div>
                    <div className="form__input">
                        <label>Location</label>
                        <select className="select" onChange={this.onLocationChange} value={this.state.location}>
                            <option key="home" value="home">Home</option>
                            <option key="away" value="away">Away</option>
                            <option key="either" value="either">Either</option>
                        </select>
                    </div>
                    <div className="form__input">
                        <label>Start Date</label>
                        <SingleDatePicker 
                            date={this.state.startDate}
                            onDateChange={this.onDateChange("start")}
                            focused={this.state.startCalendarFocused}
                            onFocusChange={this.onCalendarFocusChanged("start")}
                            numberOfMonths={1}
                            isOutsideRange={()=> false}
                            block
                        />
                    </div>
                    <div className="form__input">
                        <label>End Date</label>
                        <SingleDatePicker 
                            date={this.state.endDate}
                            onDateChange={this.onDateChange("end")}
                            focused={this.state.endCalendarFocused}
                            onFocusChange={this.onCalendarFocusChanged("end")}
                            numberOfMonths={1}
                            isOutsideRange={()=> false}
                            block
                        />
                    </div>
                    <div className="buttons">
                        <button className="button" onClick={this.resetState}>Reset</button>
                        <button className="button">Submit</button>
                    </div>
                </form>
            </div>
        )
    }

}


export default SearchForm;