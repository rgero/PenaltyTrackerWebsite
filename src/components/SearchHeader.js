import React from 'react';

const SearchHeader = () => (
    <div className="page-header">
        <h1 className="page-header__title">
            Enter your search criteria
        </h1>
        <div className=".page-header__description">
            Please fill out all of the members you'd like. If a field is blank, all members will be used.
        </div>
    </div>
);

export default SearchHeader;
