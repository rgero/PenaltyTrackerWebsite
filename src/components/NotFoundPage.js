import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';

const NotFoundPage = () => (
  <div>
    <Header/>

    <div className="content-container">
      <div className="content-container__title">
        Page Not Found
      </div>
      <div>
      The page you are trying to access does not exist, please return <Link to="/">home</Link>
      </div>
    </div>
  </div>
);

export default NotFoundPage;
