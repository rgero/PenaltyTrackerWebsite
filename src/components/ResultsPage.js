import React, { Component } from 'react';
import {connect} from 'react-redux';

//Components
import Header from './Header';

export class ResultsPage extends Component {

  constructor(props){
      super(props);
      if (!props.query){
        this.props.history.push('/')
      }
  }

  render() {
    return (
      <div>
        <Header/>
        RESULTS!
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
    return {
        query: state.query,
        results: null
    };
}

export default connect(mapStateToProps)(ResultsPage);
