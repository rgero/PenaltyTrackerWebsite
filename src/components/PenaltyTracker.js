import React from 'react';
import {connect} from 'react-redux';
//Components
import Header from './Header';
import SearchHeader from './SearchHeader';
import { SearchForm } from './SearchForm';
import {executeQuery} from '../actions/penalties';

export class PenaltyTracker extends React.Component {

  constructor(props){
    super(props);
    this.onSubmit = this.onSubmit.bind(this);

  }

  onSubmit(query){
    this.props.onSubmit(query);
    this.props.history.push('/results')
  }

  render() {
    return (
      <div>
        <Header/>
        <SearchHeader/>
        <SearchForm onSubmit={this.onSubmit}/>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onSubmit: (query) => dispatch(executeQuery(query))
  }
}

export default connect(undefined, mapDispatchToProps)(PenaltyTracker);
