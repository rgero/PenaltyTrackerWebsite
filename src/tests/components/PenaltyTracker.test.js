import React from 'react'
import {shallow} from 'enzyme'
import {PenaltyTracker} from '../../components/PenaltyTracker';

test("PenaltyTracker Rendering Test", ()=>{
    const wrapper = shallow(<PenaltyTracker/>);
    expect(wrapper).toMatchSnapshot();
})