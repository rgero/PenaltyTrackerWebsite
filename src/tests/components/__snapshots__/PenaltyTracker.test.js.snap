// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PenaltyTracker Rendering Test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PenaltyTracker />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchHeader />,
        <SearchForm
          onSubmit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchHeader />,
          <SearchForm
            onSubmit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
